core/main/src/firebolt/handlers/provider_registrar.rs

```diff
@@ -292,61 +292,59 @@ impl ProviderRegistrar {
                 Ok(r) => r,
                 Err(e) => {
                     error!("callback_app_event_emitter: Error: {:?}", e);
                     return Err(Error::Custom("Missing event_data".to_string()));
                 }
             };
 
-            if let Some(event_data_map) = event_data.as_object_mut() {
-                if let Some(event_schema_map) = context
-                    .platform_state
-                    .open_rpc_state
-                    .get_openrpc_validator()
-                    .get_result_properties_schema(event)
-                {
-                    // Populate the event result, injecting the app ID if the field exists in the event schema
-
-                    let mut result_map = Map::new();
-
-                    for key in event_schema_map.keys() {
-                        if let Some(event_value) = event_data_map.get(key) {
-                            result_map.insert(key.clone(), event_value.clone());
-                        } else if key.eq("appId") {
-                            if let Some(context) = call_context.clone() {
-                                result_map.insert(key.clone(), Value::String(context.app_id));
+            let result_value = match event_data {
+                Value::Object(ref event_data_map) => {
+                    if let Some(event_schema_map) = context
+                        .platform_state
+                        .open_rpc_state
+                        .get_openrpc_validator()
+                        .get_result_properties_schema(event)
+                    {
+                        // Populate the event result, injecting the app ID if the field exists in the event schema
+
+                        let mut result_map = Map::new();
+
+                        for key in event_schema_map.keys() {
+                            if let Some(event_value) = event_data_map.get(key) {
+                                result_map.insert(key.clone(), event_value.clone());
+                            } else if key.eq("appId") {
+                                if let Some(context) = call_context.clone() {
+                                    result_map.insert(key.clone(), Value::String(context.app_id));
+                                } else {
+                                    error!("callback_app_event_emitter: Missing call context, could not determine app ID");
+                                    result_map.insert(key.clone(), Value::Null);
+                                }
                             } else {
-                                error!("callback_app_event_emitter: Missing call context, could not determine app ID");
-                                result_map.insert(key.clone(), Value::Null);
-                            }
-                        } else {
-                            error!(
+                                error!(
                                 "callback_app_event_emitter: Missing field in event data: field={}",
```

Comment:
Should this also return `Err<T>` ?

Reply:
No, because we still want to return the event_handler and emit the event, but just log that the event data isn't what was expected.